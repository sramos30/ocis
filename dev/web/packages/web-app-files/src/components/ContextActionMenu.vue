<template>
  <div id="oc-files-context-menu">
    <oc-list
      v-for="(section, sectionIndex) in menuSections"
      :id="`oc-files-context-actions-${section.name}`"
      :key="`section-${section.name}-list`"
      class="oc-files-context-actions"
      :class="getSectionClasses(sectionIndex)"
    >
      <action-menu-item
        v-for="(action, actionIndex) in section.items"
        :key="`section-${section.name}-action-${actionIndex}`"
        :action="action"
        :items="items"
        class="oc-files-context-action oc-px-s oc-rounded"
      />
    </oc-list>
  </div>
</template>

<script>
import ActionMenuItem from './ActionMenuItem.vue'

export default {
  name: 'ContextActionMenu',
  components: { ActionMenuItem },
  props: {
    menuSections: {
      type: Array,
      required: true
    },
    // items can e.g. be currentFolder (breadcrumbs) or selectedResources (resourceTable)
    items: {
      type: Array,
      required: true
    }
  },
  methods: {
    getSectionClasses(index) {
      const classes = []
      if (!this.menuSections.length) {
        return classes
      }
      if (index < this.menuSections.length - 1) {
        classes.push('oc-pb-s')
      }
      if (index > 0) {
        classes.push('oc-pt-s')
      }
      if (index < this.menuSections.length - 1) {
        classes.push('oc-files-context-actions-border')
      }
      return classes
    }
  }
}
</script>

<style lang="scss">
.oc-files-context-actions {
  text-align: left;
  white-space: normal;

  > li {
    padding: 6px;
    &:hover {
      background-color: var(--oc-color-background-hover);
    }

    a,
    button,
    span {
      color: var(--oc-color-swatch-passive-default) !important;
      display: inline-flex;
      font-weight: normal !important;
      gap: 10px;
      justify-content: flex-start;
      vertical-align: top;
      width: 100%;
      text-align: left;

      &:hover {
        color: var(--oc-color-swatch-passive-default);
        text-decoration: none !important;
      }
    }
  }

  &-border {
    border-bottom: 1px solid var(--oc-color-border);
  }
}
</style>
