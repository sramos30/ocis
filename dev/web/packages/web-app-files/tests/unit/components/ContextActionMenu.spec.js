import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import DesignSystem from 'owncloud-design-system'
import GetTextPlugin from 'vue-gettext'

import ContextActionMenu from '@files/src/components/ContextActionMenu'

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(DesignSystem)
localVue.use(GetTextPlugin, {
  translations: 'does-not-matter.json',
  silent: true
})

describe('ContextActionMenu component', () => {
  it('renders the menu with actions', () => {
    const menuSections = [{ name: 'action 1' }, { name: 'action 2' }]
    const wrapper = getShallowWrapper(menuSections)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.oc-files-context-actions').exists()).toBeTruthy()
    expect(wrapper.findAll('.oc-files-context-actions').length).toEqual(menuSections.length)
  })
})

function getShallowWrapper(menuSections, items = []) {
  return shallowMount(ContextActionMenu, {
    localVue,
    propsData: {
      menuSections,
      items
    }
  })
}
