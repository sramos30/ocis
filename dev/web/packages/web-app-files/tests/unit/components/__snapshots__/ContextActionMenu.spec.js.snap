// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextActionMenu component renders the menu with actions 1`] = `
<div id="oc-files-context-menu">
  <oc-list-stub id="oc-files-context-actions-action 1" class="oc-files-context-actions oc-pb-s oc-files-context-actions-border"></oc-list-stub>
  <oc-list-stub id="oc-files-context-actions-action 2" class="oc-files-context-actions oc-pt-s"></oc-list-stub>
</div>
`;
